include/oneapi/dpl/internal/distributed_ranges_impl/shp/algorithms/fill.hpp

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+#include <sycl/sycl.hpp>
+
+#include <oneapi/dpl/internal/distributed_ranges_impl/concepts/concepts.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/detail/segments_tools.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/detail.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/device_ptr.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/util.hpp>
+
+namespace oneapi::dpl::experimental::dr::shp
+{
+
+template <std::contiguous_iterator Iter>
+requires(!std::is_const_v<std::iter_value_t<Iter>> && std::is_trivially_copyable_v<std::iter_value_t<Iter>>) sycl::event
+    fill_async(Iter first, Iter last, const std::iter_value_t<Iter>& value)
+{
+    auto&& q = __detail::get_queue_for_pointer(first);
+    std::iter_value_t<Iter>* arr = std::to_address(first);
+    // not using q.fill because of CMPLRLLVM-46438
```

Comment:
Please remove internal ticket numbers.

Reply:
added explanation, but ticked ID left for now